Tests/UnitTests/Plugins/Geant4/Geant4SurfaceProviderTests.cpp

```diff
@@ -0,0 +1,302 @@
+  for (int nArm = 0; nArm < nArms; nArm++) {
+    for (int nChip = 0; nChip < nChips; nChip++) {
+      int sign = (nArm == 0) ? 1 : -1;
+      double posX = sign * (armOffset + nChip * cm);
+      double posY = 0, posZ = 10;
+      Acts::Vector3 pos = Acts::Vector3(posX, posY, posZ);
+
+      BOOST_CHECK_EQUAL(s1D.at(nChips * nArm + nChip)->center(gctx), pos);
+    }
+  }
+
+  // 2D selection -- select only the second row
+  // of the left arm
+  auto sp2DCfg = Acts::Experimental::Geant4SurfaceProvider<2>::Config();
+  sp2DCfg.gdmlPath = gdmlPath.string();
+
+  auto kdt2DOpt = Acts::Experimental::Geant4SurfaceProvider<2>::kdtOptions();
+  kdt2DOpt.range = Acts::RangeXD<2, Acts::ActsScalar>();
+  kdt2DOpt.range[0].set(8, 12);
+  kdt2DOpt.range[1].set(armOffset - 5, armOffset + 100);
+  kdt2DOpt.binningValues = {Acts::BinningValue::binZ};
+
+  auto sp2D = std::make_shared<Acts::Experimental::Geant4SurfaceProvider<2>>(
+      sp2DCfg, kdt2DOpt);
+
+  auto lb2DCfg = Acts::Experimental::LayerStructureBuilder::Config();
+  lb2DCfg.surfacesProvider = sp2D;
+
+  auto lb2D =
+      std::make_shared<Acts::Experimental::LayerStructureBuilder>(lb2DCfg);
+
+  auto [s2D, v2D, su2D, vu2D] = lb2D->construct(gctx);
+
+  BOOST_CHECK_EQUAL(s2D.size(), nChips);
+  for (int nChip = 0; nChip < nChips; nChip++) {
+    double posX = armOffset + nChip * cm;
+    double posY = 0, posZ = 10;
+    Acts::Vector3 pos = Acts::Vector3(posX, posY, posZ);
+
+    BOOST_CHECK_EQUAL(s2D.at(nChip)->center(gctx), pos);
+  }
+
+  // Preselect the left arm based on the position
+  // and select only the second row
+  auto sp2DPosCfg = Acts::Experimental::Geant4SurfaceProvider<1>::Config();
+  sp2DPosCfg.gdmlPath = gdmlPath.string();
+  std::map<unsigned int, std::tuple<double, double>> ranges;
+
+  std::array<unsigned int, 3> g4Axes;
+  for (auto& bv : {Acts::binX, Acts::binY, Acts::binZ}) {
+    g4Axes[bv] = Acts::binToGeant4Axis(bv);
+  }
+
+  ranges[g4Axes[0]] = std::make_tuple(armOffset - 5, armOffset + 100);
+  ranges[g4Axes[1]] = std::make_tuple(-100, 100);
+  ranges[g4Axes[2]] = std::make_tuple(-100, 100);
+
+  sp2DPosCfg.surfacePreselector =
+      std::make_shared<Acts::Geant4PhysicalVolumeSelectors::PositionSelector>(
+          ranges);
+
+  auto sp2DPos = std::make_shared<Acts::Experimental::Geant4SurfaceProvider<1>>(
+      sp2DPosCfg, kdt1DOpt);
+
+  auto lb2DPosCfg = Acts::Experimental::LayerStructureBuilder::Config();
+  lb2DPosCfg.surfacesProvider = sp2DPos;
+
+  auto lb2DPos =
+      std::make_shared<Acts::Experimental::LayerStructureBuilder>(lb2DPosCfg);
+
+  auto [s2DPos, v2DPos, su2DPos, vu2DPos] = lb2DPos->construct(gctx);
+
+  BOOST_CHECK_EQUAL(s2DPos.size(), nChips);
+  for (int nChip = 0; nChip < nChips; nChip++) {
+    double posX = armOffset + nChip * cm;
+    double posY = 0, posZ = 10;
+    Acts::Vector3 pos = Acts::Vector3(posX, posY, posZ);
+
+    BOOST_CHECK_EQUAL(s2DPos.at(nChip)->center(gctx), pos);
+  }
+
+  std::filesystem::remove(gdmlPath);
```

Comment:
I am not sure if we should remove the files later. maybe give a more descriptive name and then we could use them for debugging?